File: src/api/java/ArrayExpr.java
/**
Copyright (c) 2012-2014 Microsoft Corporation
   
Module Name:

    ArrayExpr.java

Abstract:

Author:

    @author Christoph Wintersteiger (cwinter) 2012-03-15

Notes:
    
**/

package com.microsoft.z3;


/**
 * Array expressions
 **/
public class ArrayExpr<D extends Sort, R extends Sort> extends Expr<ArraySort<D, R>>
{
    /**
     * Constructor for ArrayExpr
     **/
    ArrayExpr(Context ctx, long obj)
    {
        super(ctx, obj);
    }
}

File: src/api/dotnet/ArrayExpr.cs
/*++
Copyright (<c>) 2012 Microsoft Corporation

Module Name:

    ArrayExpr.cs

Abstract:

    Z3 Managed API: Array Expressions

Author:

    Christoph Wintersteiger (cwinter) 2012-11-23

Notes:
    
--*/
using System.Diagnostics;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;


namespace Microsoft.Z3
{
    /// <summary>
    /// Array expressions
    /// </summary>
    public class ArrayExpr : Expr
    {
        #region Internal
        /// <summary> Constructor for ArrayExpr </summary>
        internal ArrayExpr(Context ctx, IntPtr obj)
            : base(ctx, obj)
        {
            Debug.Assert(ctx != null);
        }
        #endregion	

        /// <summary> Single argument select </summary>
        public Expr this[Expr index] 
        {
            get { return Context.MkSelect(this, index); }
        }

        /// <summary> Multi argument select </summary>
        public Expr this[IEnumerable<Expr> index] 
        {
            get { return Context.MkSelect(this, index.ToArray()); }
        }

    }
}
